### DISCUSSION TITLE
convert response struct to specific object type?

### DISCUSSION THREAD

**Discussion by rjhornsby** [aws/aws-sdk-ruby#2615] [Q&A]
I've got some Terraform code, but no state file. I'm working to match up the TF declared resources with existing AWS resources. More or less, I'm doing this with the Ruby SDK:

```
rds_client = Aws::RDS::Client.new(region: 'us-west-2', profile: 'credential-profile')
instances = rds_client.describe_db_instances
```

The response is a struct, but not an object of type `AWS::RDS::DBInstance`. Is it possible to easily take an arbitrary response struct and convert it to a specific (ie RDS::DBInstance) object class? Or more specifically, within the struct is the pagination `marker` and an array of hashes that contain the db instance metadata.

Part of the issue I'm running into with the struct is writing code that handles different AWS resource types somewhat generically. The `RDS::DBInstance` class has an `id` method, but the struct requires you to know that the property is `db_instance_identifier` - which will be different for every AWS resource type.

Alternatively, is there a data file somewhere that maps AWS resources, for example for AWS::RDS::DBInstance, the struct key pointing at the array of data is `db_instances` and the id property is `db_instance_identifier` I've tried looking at the SDK code, but nothing jumps out at me as being obvious and at least a little generic so I can use the same general approach for EC2 instances, RDS, etc. thanks!

**Comment by mullermp** [comment]
Thanks for opening up a discussion. `Aws::RDS::DBInstance` is a "resource" class, instantiated from a Resource client. You can use `resource_client = Aws::RDS::Resource.new` and `resource_client.db_instances` to call `describe_db_instances` and it will return an instance of `Aws::RDS::DBInstance`. You can also construct it manually using the result of `describe_db_instances` by iterating instances in `db_instances` and passing `:data` and `:id`:

```
          batch = []
          page.data.db_instances.each do |d|
            batch << DBInstance.new(
              id: d.db_instance_identifier,
              data: d,
              client: @client
            )
          end
```
